PhysicsTools/NanoAOD/python/custom_run3scouting_cff.py

```diff
@@ -0,0 +1,17 @@
+import FWCore.ParameterSet.Config as cms
+from PhysicsTools.NanoAOD.run3scouting_cff import *
+
+particleTableTask = cms.Task(scoutingPFCands, particleScoutingTable)
+
+ak4JetTableTask = cms.Task(ak4ScoutingJets,ak4ScoutingJetParticleNetJetTagInfos,ak4ScoutingJetParticleNetJetTags,ak4ScoutingJetTable)
+
+ak8JetTableTask = cms.Task(ak8ScoutingJets,ak8ScoutingJetsSoftDrop,ak8ScoutingJetsSoftDropMass,ecfNbeta1,Njettiness,ak8ScoutingJetParticleNetJetTagInfos,ak8ScoutingJetParticleNetJetTags,ak8ScoutingJetParticleNetMassRegressionJetTags,ak8ScoutingJetJetTable)
+
+nanoTableTaskCommon = cms.Task(photonScoutingTable,muonScoutingTable,electronScoutingTable,trackScoutingTable,primaryvertexScoutingTable,displacedvertexScoutingTable,rhoScoutingTable,metScoutingTable,particleTableTask,ak4JetTableTask,ak8JetTableTask)
+
+nanoSequenceCommon = cms.Sequence(nanoTableTaskCommon)
+
+nanoSequence = cms.Sequence(nanoSequenceCommon)
```

Comment:
@vlimant is this similar to what you had in mind when you said to create a plain cff?

Reply:
good indeed

Reply:
@vlimant there is one thing I am struggling with in this new approach of a plain cff. Maybe you happen to know a way around it?

For nanoSequenceMC there are two tasks in addition to nanoSequenceCommon. In the past I added them like [this](https://github.com/alintulu/cmssw/blob/05a945895140f5a9adf2fe17cf05566e710e8799/PhysicsTools/NanoAOD/python/custom_run3scouting_cff.py#L525-L553). Which in this new approach would translate to something like

```python
genJetTask = cms.Task(ak4MatchGen, ak8MatchGen)
nanoSequenceMC = cms.Sequence(nanoSequenceCommon + cms.Sequence(genJetTask))
```

However this would only run the two EDProducers and I would still need to add the results as external variables to the AK4 and AK8 jet table. In the old approach it was achieved with

```python
externalVariables = getattr(process.ak8JetTable, 'externalVariables', cms.PSet())
externalVariables.genJetAK8Idx = ExtVar(cms.InputTag("ak8MatchGen"), int, doc="gen jet idx")
process.ak8JetTable.externalVariables = externalVariables
```

Which was fine as these lines would only be called if the input [dataset was MC](https://github.com/alintulu/cmssw/blob/05a945895140f5a9adf2fe17cf05566e710e8799/PhysicsTools/NanoAOD/python/custom_run3scouting_cff.py#L580-L583).  However now with the plain cff, I am not sure how I can achieve this as there isn't a `nanoAOD_customizeCommonMC` and `nanoAOD_customizeCommonData` but only `nanoAOD_customizeCommon`. I.e. I cannot distinguish between if the input dataset is MC or data. I hope it is clear where I am having an issue.

Would you happen to have any ideas? The best I can think of is to create a copy of both the AK4 and AK8 table, so that there are two separate AK4(8)JetTable tasks, one which is called in nanoSequence and one which is called in nanoSequenceMC.

Reply:
hello, removing the distinction between MC and data in nano configuration was on the wish-list and done recently #40682 
the suggestion would be to have a table extension for the MC specific content, add in the MC sequence only

Reply:
to be more concrete make `ak8ScoutingJetJetGenTable` module with `extension=True` and `name=ScoutingFatJet` to include the `genJetAK8Idx` external variable

Reply:
I think I got it, thanks! I will implement it that way.